### ISSUE TITLE
500 error when called from Plex-Trakt-Scrobbler

### ISSUE THREAD

**Issue by STJBaxter** [trakt/api-help#340]
I have been getting 500 server errors for two days when performing a sync using Plex-Trakt-Scrobbler

**Comment by rectifyer** [comment]
No, I would need to exact API method that isn’t working for you and ideally a full API request and response including headers.

**Comment by STJBaxter** [comment]
I'm struggling to find the full API request in the logs, but this is what the scrobbler is trying to do when it fails:

2022-04-21 14:25:40,096 - trakt.interfaces.base            (7f67c3347b38) :  WARNING (trakt.interfaces.base:48) - Request failed: "GET /sync/playback/movies" - 500: "Server error" (Server Error)

**Comment by 123Venom** [comment]
Quite a few Kodi addons with 500 status_codes being returned and seems consistent with `sync/playback/` endpoint as the cause.  Here's two....

```
2022-04-21 17:18:37.309 T:5192     INFO <general>: SEREN (-1): Failed to sync activity: Movie bookmarks - HTTPSConnectionPool(host='api.trakt.tv', port=443): Max retries exceeded with url: /sync/playback/movies?extended=full&limit=50&page=1 (Caused by ResponseError('too many 500 error responses'))
```

```
[2022-04-21 15:54:47] [COLOR red][ Venom: ERROR ][/COLOR]: From func name: \resources\lib\modules\trakt.py.getTrakt() Line # :43
                       msg : getTrakt Error:  -> RetryError -> HTTPSConnectionPool(host='api.trakt.tv', port=443): Max retries exceeded with url: /sync/playback/?extended=full (Caused by ResponseError('too many 500 error responses'))
```

**Comment by rectifyer** [comment]
I pushed a potential fix for this, please let me know if that method works now.

**Comment by 123Venom** [comment]
@rectifyer ...fingers crossed here but both those addons I mentioned ran a successful sync moments ago and neither returned a status_code of 500 now.  They were throwing that all day long.  Will keep an eye on it but looks good atm.

**Comment by sharkykh** [comment]
@rectifyer 
My issue seems to be somewhat related to this, so I'll comment here instead of a new issue.
I seem to be getting CORS errors - __browser__ preflight requests error out with HTTP 502.

For example, a request to exchange a token fails with:
`Access to fetch at 'https://api.trakt.tv/oauth/token' from origin 'http://localhost:8080' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

The `http://localhost:8080` origin __is set__ on my API app (on Trakt.tv).
My app was working fine (for me), but as part of debugging another user's issue,
I disconnected the app _from my **account**_ and attempted to reauthorize it, _then_ I started getting CORS errors.

I would appreciate if could look into it, if you require any more info from me I can provide whatever you need.

**Comment by STJBaxter** [comment]
> @rectifyer ...fingers crossed here but both those addons I mentioned ran a successful sync moments ago and neither returned a status_code of 500 now. They were throwing that all day long. Will keep an eye on it but looks good atm.

It does indeed look like you fixed that 500 error - thanks for doing it so quickly, I can see that my sync has been working overnight.

I am also now experiencing the 502 error mentioned above however:
2022-04-22 08:39:45,781 - trakt.interfaces.base            (7f67c3347b38) :  WARNING (trakt.interfaces.base:48) - Request failed: "GET /sync/last_activities" - 502: "Server unavailable" (Bad Gateway)
